### PR TITLE
pdal: 2.7.2 -> 2.8.0

### PR DESCRIPTION
**Pull request by imincik**
## Description of changes

Update PDAL to the latest version.

Release notes:
https://github.com/PDAL/PDAL/releases/tag/2.8.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/pdal/default.nix [hunk 1]
```diff
@@ -28,19 +28,15 @@
...
   pname = "pdal";
-  version = "2.7.2";
+  version = "2.8.0";
 
   src = fetchFromGitHub {
     owner = "PDAL";
     repo = "PDAL";
     rev = finalAttrs.version;
-    sha256 = "sha256-ukBZLr/iyYQ68sv9JWrR4YP0ahHfGhytgcWKPzrF3Ps=";
+    sha256 = "sha256-VHcUyYADmNzxZ+Ix56TNuaP9wG+vRlEl71cNM0uMaHg=";
```
Comment (sikmir) [inline pkgs/development/libraries/pdal/default.nix:37]:
```suggestion
    hash = "sha256-VHcUyYADmNzxZ+Ix56TNuaP9wG+vRlEl71cNM0uMaHg=";
```

File: pkgs/development/libraries/pdal/default.nix [hunk 2]
```diff
@@ -28,19 +28,15 @@
...
   };
 
-  patches = [
-    ./pdal.pc.in.patch
```
Comment (sikmir) [inline pkgs/development/libraries/pdal/default.nix:41]: Fixed in https://github.com/PDAL/PDAL/pull/4464

**Comment by imincik** [comment]
Result of `nixpkgs-review pr 340015` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>entwine</li>
    <li>grass</li>
    <li>pdal</li>
    <li>qgis</li>
    <li>qgis-ltr</li>
  </ul>
</details>

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr -P qgis -P qgis-ltr 340015` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>grass</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pdal</li>
  </ul>
</details>

**Review by sikmir — APPROVED** [review]

**Review by l0b0 — APPROVED** [review]
LGTM; untested.

**Comment by imincik** [comment]
> grass is broken on darwin on master:
> 
> ```shell
> grass> patching file db/drivers/mysql/db.c
> grass> Hunk #1 FAILED at 52.
> grass> 1 out of 1 hunk FAILED -- saving rejects to file db/drivers/mysql/db.c.rej
> ```

Thanks, issue created https://github.com/NixOS/nixpkgs/issues/340475

**Pull request merged**